Projects/Backendless.Xamarin/LocalStorage.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using Plugin.DeviceInfo;
+
+namespace BackendlessAPI.Xamarin
+{
+  public class LocalStorage
+  {
+    private String path = System.Environment.GetFolderPath( System.Environment.SpecialFolder.LocalApplicationData, Environment.SpecialFolderOption.Create );
+    private Boolean _hasData;
+    private String _deviceId;
+
+    public LocalStorage()
+    {
+      HasData = !String.IsNullOrEmpty( GetData() );
+    }
+
+    public Boolean HasData
+    {
+      get => _hasData;
+      private set => _hasData = value;
+    }
+
+    public String DeviceId
+    {
+      get
+      {
+        if( HasData )
+          return _deviceId;
+        else
+        {
+          SaveData();
+          return _deviceId = GetData();
+        }
+      }
+      private set => _deviceId = String.IsNullOrEmpty( _deviceId ) ? value : _deviceId;
+    }
+
+    public void SaveData()
+    {
+      try
+      {
+        String fileName = Path.Combine( path, "BackendlessDeviceId.txt" );
```

Comment:
Why does a class with a generic name "LocalStorage" is hardcoded to worj with "BackendlessDeviceId.txt ?
Why do we need this class?

Reply:
These are unnecessary files. Have been removed.